case-lib/hijack.sh

```diff
@@ -50,7 +50,12 @@ function exit()
 
     # check if function already defined
     # on exit check whether pulseaudio is disabled
-    [[ $(declare -f func_lib_restore_pulseaudio) ]] && func_lib_restore_pulseaudio
+    [[ $(declare -f func_lib_restore_pulseaudio) ]] && ret=func_lib_restore_pulseaudio
+    # if test case passed, but failed to restore pulseaudio, set exit_status to ret,
+    # this helps to dectect pulseaudio failures.
+    if [ $exit_status -eq 0 -a $ret -ne 0]; then
```

Comment:
you logic is when case failed, it will continue disable pulseaudio  
but I think it is not refer the test-case  
it should be when case run finished, restore pulseaudio